addons/mail/models/mail_activity.py

```diff
@@ -127,9 +127,27 @@ def default_get(self, fields):
         'Document Name', compute='_compute_res_name', compute_sudo=True, store=True,
         help="Display name of the related document.", readonly=True)
     # activity
+    @api.model
+    def default_get_activity_type(self):
+        activity_type = self.env["mail.activity.type"]
+        ir_model_obj = self.env['ir.model']
+        current_model_name = self.env.context['default_res_model']
+        current_model = ir_model_obj.search([('model', '=', current_model_name)])
+        current_model_id = current_model[0].id
```

Comment:
Here you can use self.env['ir.model']._get(model_name).id . It is a  more elegant shortcut.